src/toil/common.py

```diff
@@ -524,3 +525,33 @@ def parseSetEnv(l):
             raise ValueError('Empty name')
         d[k] = v
     return d
+
+
+def getToilWorkDir(configWorkDir=None):
+    """
+    Returns a path to the directory where worker directories and the cache will be located.
+    This is particularly useful if the user hasn't provided a value for --workDir (passed to this
+    function as configWorkDir) and the TOIL_WORKDIR environment variable is unset.  In this case,
+    the value returned points to a directory 'toil' located somewhere on the system in a location
+    reserved for temporary files (obtained from the tempfile module).  The variable is created if
+    required.
+    Hierarchy for workdir is config.workDir >> $TOIL_WORKDIR >> $TMPDIR
+    :param configWorkDir: Value passed to the program using the workDir flag
+    :return: Path to the work dir
+    """
+    configWorkDir = configWorkDir or os.getenv('TOIL_WORKDIR')
+    if configWorkDir is not None and not os.path.exists(configWorkDir):
+        raise RuntimeError("The temporary directory specified by workDir: %s does not "
+                           "exist" % tempRootDir)
+    elif configWorkDir is None:
+        #  Neither --workDir not TOIL_WORKDIR is set
+        configWorkDir = os.path.join(tempfile.gettempdir(), 'toil')
```

Comment:
yes, it's worthy of discussion. A user *can* really specify a non-empty directory as the workdir.